riscv/mmu.cc

```diff
@@ -204,21 +204,21 @@ reg_t mmu_t::pmp_ok(reg_t addr, reg_t len, access_type type, reg_t mode)
     return true;
 
   reg_t base = 0;
-  for (size_t i = 0; i < proc->state.n_pmp; i++) {
-    reg_t tor = proc->state.pmpaddr[i] << PMP_SHIFT;
+  for (size_t i = 0; i < (size_t)proc->state.platform_n_pmp; i++) {
+    reg_t tor = proc->state.pmpaddr[i] << proc->state.pmp_granularity;
```

Comment:
I don't believe this new logic is correct.  PMP_SHIFT is a constant no matter what the granularity is.  The granularity just changes the range of values that the register can hold.

Reply:
Err, I meant to say, "the granularity just changes how the comparators behave".  See the Rocket source code I referenced in the other comment.  The granularity does not affect how stuff gets stored in the register, how much it's shifted, etc.